graph/appender/sidecars_check.go

```diff
@@ -48,38 +50,26 @@ func (a *SidecarsCheckAppender) applySidecarsChecks(trafficMap graph.TrafficMap,
 			continue
 		}
 
-		// get the pods for the node, either by app+version labels, or workload deployment
-		var err error
-		var pods models.Pods
+		// get the workloads for the node and check to see if they have sidecars
+		hasIstioSidecar := false
 		switch n.NodeType {
 		case graph.NodeTypeWorkload:
-			workload, err := business.Workload.GetWorkload(n.Namespace, n.Workload, false)
-			if err == nil {
-				pods = workload.Pods
-			}
+			workload := getWorkload(n.Workload, workloadList)
+			hasIstioSidecar = workload != nil && workload.IstioSidecar
 		case graph.NodeTypeApp:
-			podLabels := a.getAppLabels(appLabel, n.App, versionLabel, n.Version)
-			pods, err = business.Workload.GetPods(n.Namespace, podLabels)
+			workloads := getAppWorkloads(n.App, n.Version, workloadList)
```

Comment:
GetApp / GetAppList logic is calculating the sidecars for you.
Better to use them (and modify those methods in what we need) instead to calculate if istioSidecar is present in different places; that info should come from the business layer.